sqlserver/datadog_checks/sqlserver/utils.py

```diff
@@ -35,9 +41,34 @@ def set_default_driver_conf():
         # Use default `./driver_config/odbcinst.ini` when Agent is running in docker.
         # `freetds` is shipped with the Docker Agent.
         os.environ.setdefault('ODBCSYSINI', DRIVER_CONFIG_DIR)
-    else:
+    elif Platform.is_linux():
+        """
+        The agent running on Linux has msodbcsql18 and FreeTDS installed.
+        The default driver is msodbcsql18.
+        To best leverage the default driver, we set the ODBCSYSINI environment variable to the directory
+        containing the pre-configured odbcinst.ini file.
+        However, if the user has already configured the ODBCSYSINI environment variable,
+        OR if the user has already created or copied the odbcinst.ini file in the unixODBC sysconfig location,
+        we do not override the ODBCSYSINI environment variable.
+        """
+        if pyodbc is None:
+            return
+
+        if 'ODBCSYSINI' in os.environ:
+            # If ODBCSYSINI is already set in env, don't override it
+            return
+
+        if pyodbc.drivers() or pyodbc.dataSources():
```

Comment:
`pyodbc.drivers()` triggers unixODBC loads the drivers into memory. Even when no driver was found and the program sets the `ODBCSYSINI` in the environment variable, unixODBC will not reload the driver info from the config dir from `ODBCSYSINI`. 
This prevents pyodbc from loading the driver file and causing driver not found error. Reloading pyodbc package using `importlib.reload(pyodbc)` does not reload the drivers either. 
However this line of code is critical for backward compatibility purpose. If user is using a self-installed odbc driver (e.g. msodbcsql 17), we don't want to force the upgrade to agent embedded msodbcsql18. This might cause issue with legacy sqlserver databases.